### PR TITLE
Sendgrid template support

### PR DESCRIPTION
**Pull request by leantrace**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -39,6 +39,24 @@ The plugin only takes two configuration parameters:
 .. image:: docs/config.png
     :alt: Configuration
 
+Sendgrid Templating
+"""""""""""""
+To activate Sendgrid Templated, override the desired email *.txt.vm template as documented in Curity and add at least one mandatory parameter:
+``sendgridTemplateId=1234abc``
+You can as many additional parameters as well, which will be passed as sendgrid parameters to the template and substituted in the template.
```
Comment (anestos) [inline README.rst]: You can **_have_**

File: src/test/com/example/curity/SendGridEmailSender/SendgridEmailSenderTest.groovy [hunk 2]
```diff
@@ -15,6 +15,8 @@
  */
 package com.example.curity.SendGridEmailSender
```
Comment (anestos) [inline src/test/com/example/curity/SendGridEmailSender/SendgridEmailSenderTest.groovy]: suggest to change the package here too

**Review by anestos — COMMENTED** [review]
I would rather make this a bit more strict. 
Namely, add a flag in the configuration, i.e `send-templated-emails` which will be `true/false`. 
When false, the previous code should run as is. 
When true, the code should **enforce** that the template contains `Properties` (i.e throw an exception if the parsing in line 55 falis) and also it should eforce that the `sendgridTemplateId` is part of the properties and is an integer.

**Review by anestos — COMMENTED** [review]
_removed duplicate comment_

File: src/test/com/example/curity/SendGridEmailSender/SendgridEmailSenderTest.groovy (hunk 2 above)
Comment (leantrace) [reply src/test/com/example/curity/SendGridEmailSender/SendgridEmailSenderTest.groovy]: done

**Comment by leantrace** [comment]
> I would rather make this a bit more strict.
> Namely, add a flag in the configuration, i.e `send-sendgrid-tempalted-emails` which will be `true/false`.
> When false, the previous code should run as is.
> When true, the code should **enforce** that the template contains `Properties` (i.e throw an exception if the parsing in line 55 falis) and also it should eforce that the `sendgridTemplateId` is part of the properties and is an integer.

hey @anestos I corrected 2 of 3 points as suggested. For this one I have questions. What would be the benefit to add additional true/false parameter. Looks like redundancy to me. I mean if I have a `sendgridTemplateId ` I can assume, that it is a sendgrid template configuration following, so this parameter reflects `send-sendgrid-tempalted-emails=true/false` already.
The sendgridTemplateId is not an integer and usually looks like this: d-9de41780672e445184f4eabb907602d6

But feel free to add any idia here

**Comment by anestos** [comment]
The main benefit is to make sure that the administrator that configures the emailer knows what it is doing.
When the templating is enabled, and say they don't have a `sendgridTemplateId` we can throw an exception informing them that the intended use of the emailer is with this template and they have "missconfigured" the emailer. 
They can still have two instances of the emailer, one with templated true and one with false for different use-cases within Curity but it makes the usage clear.

**Comment by anestos** [comment]
When i referenced an integer i was talking about `sendgridAsmId` sorry

**Comment by leantrace** [comment]
hey Nikos, the `sendgridAsmId`is not mandatory and email sender works just fine without the asmId. So I preferred here more a "fail tolerate" implementation with a warning in the log. The user still gets an email in this case and knows it generally works, but might no see the asmId set (refers: email-group like 'unsubscribes'), but then see it in the log.
```
_logger.warn("Invalid ASM ID, value must be an integer, but currently is={}", p.getProperty("sendgridAsmId"));
```
In case of throwing an error as you suggest the user will not get any email and most likely look into the `sendgridTemplateId` and see that its just fine and might wonder, why it does not work.

Regarding the boolean parameter. I suspect we speak of two different things. Do you mean the property should be part of the *.txt.vm file as an first parameter, something like this?
```
send-sendgrid-tempalted-emails=true
sendgridTemplateId=123-abc
```
Then what for? I mean if someone adds the `sendgridTemplateId ` then I guess this person knows what she/he is doing.

Or do you mean in the UI? 

Long story short:
- Feel free to make it more strict with asmId
- Feel free to add the configuration `send-sendgrid-tempalted-emails`


I think its  quicker

**Review by gary-archer — APPROVED** [review]
